libs/newsletter/src/driver/hubspot/newsletter.service.spec.ts

```diff
@@ -0,0 +1,139 @@
+import { TestBed, async, inject } from '@angular/core/testing';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+
+import { DaffNewsletterConfig } from '../injection-tokens/newsletter-config.token';
+import { DaffNewsletterTransformer } from '../injection-tokens/newsletter-transformer.token';
+import { DaffNewsletterHubspotTransformer } from './transformers/newsletter.transformer';
+
+import { Title } from '@angular/platform-browser';
+import { HttpClient, HttpRequest } from '@angular/common/http';
+import { DaffNewsletterHubspotService } from './newsletter.service';
+import { DOCUMENT } from '@angular/common';
+import { Router } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+
+describe('DaffNewsletterHubspotService', () => {
+  let newsletterService: DaffNewsletterHubspotService;
+  let httpMock: HttpTestingController;
+  let doc: Document;
+  let httpClient: HttpClient;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule,
+        RouterTestingModule,
+      ],
+      providers: [
+        DaffNewsletterHubspotService,
+        { provide: DaffNewsletterConfig, useValue: {} },
+        { provide: DaffNewsletterTransformer, useExisting: DaffNewsletterHubspotTransformer }
+      ]
+    });
+
+    httpMock = TestBed.get(HttpTestingController);
+    newsletterService = TestBed.get(DaffNewsletterHubspotService);
+    doc = TestBed.get(DOCUMENT);
+    httpClient = TestBed.get(HttpClient);
+  });
+  afterEach(() => {
+    httpMock.verify();
+    document.cookie = 'hubspotutk=; expires=Thu, 01 Jan 1970 00:00:00 GMT'
+  });
+  it('should be created', () => {
+    expect(newsletterService).toBeTruthy();
+  });
+
+  describe('sending a newsletter submission to hubspot', () => {
+    it('should take Hubspot configuration from the module configuration', async(inject([HttpClient, DaffNewsletterConfig], (http, config) => {
+      spyOn(http, 'post').and.callThrough();
+      const submission = { email: 'email@email.com', firstName: 'John', lastName: 'Doe' };
+      const version = config.version === undefined ? 'v3' : config.version;
```

Comment:
What does this code do?

Reply:
This code generates whats the URL is suppose to look like. So line 51 is logic to make sure a version is included in the mock submission URL.

Reply:
Can we instead provide an explicit configuration object for `DaffNewsletterConfig ` instead of having some unnecessary testing logic?

Reply:
Addressed in 94d09a46797258af1bb71074d88eb9d7ca62fc86